### ISSUE TITLE
Pandas syntax incompatibility

### ISSUE THREAD

**Issue by Esfo** [xonsh/xonsh#4005]
## xonfig

<details>

```
+------------------+-----------------+
| xonsh            | 0.9.24          |
| Python           | 3.9.0           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 3.0.8           |
| shell type       | prompt_toolkit  |
| pygments         | 2.7.2           |
| on posix         | True            |
| on linux         | True            |
| distro           | arch            |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
| on jupyter       | True            |
| jupyter kernel   | None            |
+------------------+-----------------+
```

</details>

## Expected Behavior

Functional pandas syntax for referencing dataframe coordinates, as is in python.

## Current Behavior

xonsh can't seem to handle pandas syntax since I switched to python 3.9. I get similar behavior when referencing a column of an imported dataframe. Such as `t.loc[:,'hi'] * 4`
### Traceback (if applicable)

running xonsh/0.9.24 in linux
<details>

```xonsh
sfo@box ~ $ t.loc[:,'hi'] = hi
...........
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/usr/lib/python3.9/site-packages/xonsh/ptk_shell/shell.py", line 238, in _push
    code = self.execer.compile(src, mode="single", glbs=self.ctx, locs=None)
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 23992, in compile
    tree = self.parse(input, ctx, mode=mode, filename=filename, transform=transform)
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 23952, in parse
    tree, input = self._parse_ctx_free(input, mode=mode, filename=filename)
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 24095, in _parse_ctx_free
    raise original_error from None
  File "/usr/lib/python3.9/site-packages/xonsh/__amalgam__.py", line 24076, in _parse_ctx_free
    tree = self.parser.parse(
  File "/usr/lib/python3.9/site-packages/xonsh/parsers/base.py", line 501, in parse
    check_contexts(tree)
  File "/usr/lib/python3.9/site-packages/xonsh/parsers/context_check.py", line 58, in check_contexts
    raise e
  File "<string>", line None
SyntaxError: can't assign to function call
```

</details>

## Steps to Reproduce

```xonsh
sfo@box ~ $ import pandas as pd
sfo@box ~ $ hi = list(range(4))
sfo@box ~ $ t = pd.DataFrame()
sfo@box ~ $ t.loc[:]
Empty DataFrame
Columns: []
Index: []
sfo@box ~ $ t.loc[:,'hi'] = hi
...........
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
  File "<string>", line None
SyntaxError: can't assign to function call
```
## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by anki-code** [comment]
I can repeat:
```python
docker run --rm -it python:3.9 /bin/bash -c! pip install pandas git+https://github.com/xonsh/xonsh ; xonsh -c 'import pandas as pd; df = pd.DataFrame([1,2]); df.loc[:,"hi"] = 1'
# SyntaxError: can't assign to function call

# Prepared by xontrib-hist-format
```

**Comment by lypanov** [comment]
Another one. The PR is https://github.com/xonsh/xonsh/pull/3588.

**Comment by anki-code** [comment]
Fixed in master by #4101.


Tested in docker:
```python
docker run --rm -it python:3.9-slim /bin/bash -c "apt update && apt install -y git && pip install -U git+https://github.com/xonsh/xonsh.git && xonsh"
```
